app/Providers/GitHubClientServiceProvider.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Gistlog\Providers;
+
+use Github\Client as GitHubClient;
+use Github\HttpClient\CachedHttpClient as CachedGitHubClient;
+use Illuminate\Support\ServiceProvider;
+
+class GitHubClientServiceProvider extends ServiceProvider
+{
+    public function register()
+    {
+        $this->app->singleton(GitHubClient::class, function ($app) {
+
+            $githubClient = new GitHubClient(
+                new CachedGitHubClient([
+                    'cache_dir' => storage_path() . '/app/github-api-cache',
+                ])
+            );
+
+            // We're only making public API requests, so we don't *need* to
+            // authenticate, but doing so significantly increases the rate
+            // limit. So here we authenticate if credentials are provided,
+            // but if they aren't, no big deal.
+            if (config('services.github.token')) {
+                $githubClient->authenticate(
+                    config('services.github.token'),
+                    GitHubClient::AUTH_HTTP_TOKEN
```

Comment:
@mattstauffer @quickliketurtle after doing some reading, it appears that the only time we actually NEED to use token authentication with our github client is when we are performing an action on behalf of the user. In our case, that would mean the only github client call in the codebase at this moment that needs to be using the token auth is the `postGistComment` method in the `GistClient` class. Everything else should be able to use the `client id` and `client secret` to do authorizations.

The token may have been used out of convenience at some point since we had no need for the client id and secret, but at this point, they are necessary for our OAuth flow with Socialite. That being the case, I would suggest that we change all `githubClient` calls to use the `Github\Client::AUTH_URL_CLIENT_ID` method as outlined [here](https://github.com/KnpLabs/php-github-api/blob/master/doc/security.md#authenticate).

@quickliketurtle if you need to create a developer application to use for local dev, click https://github.com/settings/applications/new to create a new dev application.